pcdsdevices/state.py

```diff
@@ -692,6 +717,36 @@ class TwinCATStatePositioner(StatePositioner):
         doc='Configuration of state positions, deltas, etc.',
     )
 
+    def _get_state_velo(self, items: SignalToValue) -> float:
+        """For state_velo, calculate the velocity to show."""
+        return max(value for value in items.values())
+
+    def _set_state_velo(self, value: float) -> SignalToValue:
+        """For state_velo, distribute the puts to all fields."""
+        return {
+            getattr(self.parent, name): value for name in
+            state_config_dotted_velos(self.parent.config.state_count)
```

Comment:
I'm 50/50 on whether or not the `calculate_on_put` function we pass in should get a `SignalToValue` typed argument just like the `calculate` function does. This would make it very easy to construct the response and to change our output based on the current values.

Reply:
Ok, after thinking for 10 more seconds, I think this would be a useful addition to the API, both for the verbosity of `self.parent.stuff` and for the use cases for the information contained within

Reply:
Does `self.signals` also work in this context, actually? If so, then I think it's fine as-is... probably.

Should probably revisit at some point to add a `calculate_metadata` at least

Reply:
You're right - the `self.signals` as opposed to `self.parent.state_velo.signals` (I can't even explain what I was thinking when suggesting that) would be the appropriate shortcut.

The extra weird part in the current API is that an instance method defined in the class would normally be bound to the class instance (i.e., the `Device` here) but is instead bound to the MDS instance. So to avoid future confusion there, it may be wise to switch it up.

I like the `calculate_metadata` idea, hmm...